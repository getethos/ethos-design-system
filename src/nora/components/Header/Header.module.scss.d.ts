// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'Header': string;
  'Left': string;
  'Navbar': string;
  'Right': string;
}
export const cssExports: CssExports;
export default cssExports;
