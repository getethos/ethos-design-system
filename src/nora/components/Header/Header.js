import PropTypes from 'prop-types'
import React from 'react'
import styles from './Header.module.scss'

const Header = () => {
  return <h1 className={styles.Header}>Nora</h1>
}

Header.propTypes = {
  name: PropTypes.string.isRequired,
  title: PropTypes.string,
}

Header.defaultProps = {
  title: undefined,
}

export default Header
