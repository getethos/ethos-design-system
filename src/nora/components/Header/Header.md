A very simple header with no children:

```jsx
<Header name="nora-header-no-title" />
```

Since we used fixed position, this will cover the example from above.

```jsx
const Profile = () => {
  return <span>Profile</span>
}
;<Header leftChildren="Nora" rightChildren={Profile()} name="nora-header" />
```
