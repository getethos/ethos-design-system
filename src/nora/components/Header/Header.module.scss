@import '../Colors.css';
@import '../../../components/Spacer/Spacer.scss';

.Header {
  color: var(--NoraNeutral-400);
  font-weight: 400;
  height: var(--Space-56);
  background-color: white;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1;
  border-bottom: 1px solid var(--NoraNeutral-200);
  display: flex;
  align-items: center;
}

.Navbar {
  display: flex;
  width: 100vw;
}

.Left {
  font-size: var(--Nora-Font-15);
  flex: 1;
  margin-left: var(--Space-24);
}

.Right {
  display: flex;
  justify-content: space-between;
  margin-right: var(--Space-24);
}
