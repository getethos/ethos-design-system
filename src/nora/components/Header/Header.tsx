import * as React from 'react'
import styles from './Header.module.scss'

const Left: React.SFC<{}> = ({ children }) => {
  return <div className={styles.Left}>{children}</div>
}
Left.defaultProps = {
  children: undefined,
}

const Right: React.SFC<{}> = ({ children }) => {
  return <div className={styles.Right}>{children}</div>
}
Right.defaultProps = {
  children: undefined,
}

const Navbar: React.SFC<{}> = ({ children }) => {
  return <nav className={styles.Navbar}>{children}</nav>
}
Navbar.defaultProps = {
  children: undefined,
}

type HeaderProps = {
  name: string
  leftChildren?: React.ReactNode
  rightChildren?: React.ReactNode
}
/**
 * Base `Header` component that provides fixed positioning and renders
 * left and right side children. For ease of extension, `Header` is
 * intentionally much simpler then the UniversalNavbar component.
 * @see See [UniversalNavbar](https://github.com/getethos/ethos-design-system/blob/master/src/components/UniversalNavbar/UniversalNavbar.js)
 *
 * @public
 *
 * @param {object} props - Component Props
 * @prop {string} props.name - Unique name of header.
 * @prop {React.ReactNode} props.leftChildren - children to be rendered on left side of header
 * @prop {React.ReactNode} props.rightChildren - children to be rendered on right side of header
 *
 * @return {JSX.Element}
 */
export const Header: React.SFC<HeaderProps> = ({
  leftChildren,
  rightChildren,
}) => {
  return (
    <header className={styles.Header}>
      <Navbar>
        <Left>{leftChildren}</Left>
        <Right>{rightChildren}</Right>
      </Navbar>
    </header>
  )
}
Header.defaultProps = {
  leftChildren: undefined,
  rightChildren: undefined,
}
