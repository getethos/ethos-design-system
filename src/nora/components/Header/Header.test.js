import React from 'react'
import renderer from 'react-test-renderer'
import { Header } from './Header'

describe('Header', () => {
  it('renders defaults', () => {
    const tree = renderer
      .create(<Header name="nombre" title="Le Title" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
