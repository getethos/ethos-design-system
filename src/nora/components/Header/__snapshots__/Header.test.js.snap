// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders defaults 1`] = `
<header
  className="Header"
>
  <nav
    className="Navbar"
  >
    <div
      className="Left"
    />
    <div
      className="Right"
    />
  </nav>
</header>
`;
